ztm-front/src/components/Main.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { Sidebar } from './Sidebar'
+import { MapView } from './MapView'
+import { ButtonsList } from '../types'
+
+export const Main = () => {
+    const [currentSelect, setSelect] = useState(false)
+
+    return(
+        <MainWrapper>
+            <Sidebar
+                buttons={Object.values(ButtonsList)}
+                onSidebarSelect={() => setSelect(!currentSelect)}
```

Comment:
this is potential issue, if you click it twice it can produce bad unexpected value. useState is async please use this syntax:
```
setSelect(prevState => !prevState)
```